stdlib/string.ml

```diff
@@ -40,6 +40,7 @@ let empty = ""
 let of_bytes = B.to_string
 let to_bytes = B.of_string
 let sub s ofs len =
+  if length s = len then s else
```

Comment:
You need to check ` ofs= 0`

Reply:
Indeed, otherwise `sub s (-1) (String.length s` would return `s`, we want to have an error in this case. (I thought that I had written a review comment about this, but it looks like it was not posted.)

Reply:
It was posted here https://github.com/ocaml/ocaml/pull/13620#discussion_r1850017772

Reply:
Yes sure, that somehow works as no other value of ofs is possible, but this was a bad idea as it prevent the exception.